parser/lexer.go

```diff
@@ -527,6 +530,9 @@ func (s *Scanner) scanString() (tok int, pos Pos, lit string) {
 			ch0 = handleEscape(s)
 		}
 		mb.writeRune(ch0, s.r.w)
+		if s.r.eof() {
```

Comment:
Emmmmmmm.........I think positive logic is better than negative logic.
For loop break problem is another historical problem, maybe we should refactor it in another PR?